modules/@apostrophecms/login/index.js

```diff
@@ -981,6 +994,77 @@ module.exports = {
             required: true
           })
           );
+      },
+
+      normalizeLoginName(usernameOrEmail, {
+        caseInsensitive = self.options.caseInsensitive
+      } = {}) {
+        if (typeof usernameOrEmail !== 'string' || !caseInsensitive) {
+          return usernameOrEmail;
+        }
+        return usernameOrEmail.toLowerCase();
+      },
+
+      async addCaseInsensitiveMigration() {
+        if (self.options.caseInsensitive) {
+          self.apos.migration.add('login-case-insensitive', self.caseInsensitiveTask);
+        }
+      },
+
+      async caseInsensitiveTask() {
+        const duplicatedUsernames = [];
+        await self.apos.migration.eachDoc({ type: '@apostrophecms/user' }, 1, async (user) => {
+          const normalizedUsername = self.apos.login
+            .normalizeLoginName(user.username, { caseInsensitive: true });
+          const normalizedEmail = self.apos.login
+            .normalizeLoginName(user.email, { caseInsensitive: true });
+
+          const shouldUpdateUsername = user.username !== normalizedUsername;
+          const shouldUpdateEmail = user.email && user.email !== normalizedEmail;
+          if (!shouldUpdateUsername && !shouldUpdateEmail) {
+            return;
+          }
+
+          try {
+            if (shouldUpdateUsername) {
+              await self.apos.user.safe.updateOne(
```

Comment:
The safe has both username and email properties you need to be worrying about, but for email you should only worry about it if `email` is truthy (this is important: the email index is "unique" but also "sparse," meaning you need to leave email undefined unless the user actually has one)